pkg/logger/formatter.go

```diff
@@ -0,0 +1,271 @@
+package logger
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"runtime"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	optionalCIPathPattern              = "(_.+)?"   // allow paths like ".../<pkg>_PR-2/..."
+	optionalVersionedModulePathPattern = `(v\d+/)?` // allow paths like ".../<pkg>/v2/..."
+)
+
+var loggerFilePathPattern = fmt.Sprintf("kubectl-ces-plugin%s/pkg/%slogger/logger.go", optionalCIPathPattern, optionalVersionedModulePathPattern)
+
+// Formatter - logrus formatter, implements logrus.Formatter
+type Formatter struct {
+	// FieldsOrder - default: fields sorted alphabetically
+	FieldsOrder []string
+
+	// TimestampFormat - default: time.StampMilli = "Jan _2 15:04:05.000"
+	TimestampFormat string
+
+	// HideKeys - show [fieldValue] instead of [fieldKey:fieldValue]
+	HideKeys bool
+
+	// NoColors - disable colors
+	NoColors bool
+
+	// NoFieldsColors - apply colors only to the level, default is level + fields
+	NoFieldsColors bool
+
+	// NoFieldsSpace - no space between fields
+	NoFieldsSpace bool
+
+	// ShowFullLevel - show a full level [WARNING] instead of [WARN]
+	ShowFullLevel bool
+
+	// NoUppercaseLevel - no upper case for level value
+	NoUppercaseLevel bool
+
+	// TrimMessages - trim whitespaces on messages
+	TrimMessages bool
+
+	// CallerFirst - print caller info first
+	CallerFirst bool
+
+	// CustomCallerFormatter - set custom formatter for caller info
+	CustomCallerFormatter func(*runtime.Frame) string
+}
+
+// Format an log entry
+func (f *Formatter) Format(entry *logrus.Entry) ([]byte, error) {
```

Comment:
🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'Format' is too long (80 > 60)